### PR TITLE
Improving `isVersionMavenIncompatible` check

### PR DESCRIPTION
**Pull request by elect86**
As titled

### CODE REVIEW

**Comment by stephengold** [comment]
Looks good to me!

**Comment by stephengold** [comment]
This PR needs a milestone. Who assigns milestones?

File: subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java [hunk 1]
```diff
@@ -386,10 +386,10 @@ private boolean isVersionMavenIncompatible(String version) {
         if (version == null) {
             return false;
         }
-        if (version.contains("+")) {
+        if (version.endsWith(".+")) {
             return true;
         }
-        if (version.contains("latest")) {
+        if (version.startsWith("latest.")) {
```
Comment (donat) [inline subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java:392]: ❌ Missing integration test coverage.

**Review by donat — CHANGES_REQUESTED** [review]

**Comment by stephengold** [comment]
The issue this PR was intended to address (#16555) was recently closed by PR #17267. Perhaps this PR is no longer needed.

**Comment by elect86** [comment]
It's not, indeed

**Pull request closed without merging**